zookeeper-server/src/main/java/org/apache/zookeeper/server/ServerCnxn.java

```diff
@@ -62,13 +62,6 @@
 
     private Set<Id> authInfo = Collections.newSetFromMap(new ConcurrentHashMap<Id, Boolean>());
 
-    /**
-     * If the client is of old version, we don't send r-o mode info to it.
-     * The reason is that if we would, old C client doesn't read it, which
-     * results in TCP RST packet, i.e. "connection reset by peer".
-     */
-    boolean isOldClient = true;
```

Comment:
This workaround is introduced at [ZOOKEEPER-784](https://issues.apache.org/jira/browse/ZOOKEEPER-784). And after 3.4.0 both the server & the client handle `readOnly` field.

However, as jute's `serialize` & `deserialize` implemented atomically, this PR should be logically break client version less than 3.4 as documented here. I'll give it a test and wonder what the version policy and compatibility promise we should keep.

cc @eolivelli @phunt

Reply:
Confirm work start from 3.4.0, not for 3.3.6 and before.